storage_backblaze.go

```diff
@@ -0,0 +1,193 @@
+			repositoryFile: bucketPrefix[1] + "-repository",
+			// bucketPrefix:   bucketPrefix[1],
+			bucket:    bucket,
+			backblaze: cl,
+		}, nil
+	}
+	return &StorageBackblaze{
+		url:            URL,
+		chunkFile:      bucketPrefix[1] + "-chunks",
+		snapshotFile:   bucketPrefix[1] + "-snapshots",
+		repositoryFile: bucketPrefix[1] + "-repository",
+		// bucketPrefix:   bucketPrefix[1],
+		bucket:    bucket,
+		backblaze: cl,
+	}, nil
+
+}
+
+// Location returns the type and location of the repository
+func (backend *StorageBackblaze) Location() string {
+	return backend.url.String()
+}
+
+// Close the backend
+func (backend *StorageBackblaze) Close() error {
+	return nil
+}
+
+// Protocols returns the Protocol Schemes supported by this backend
+func (backend *StorageBackblaze) Protocols() []string {
+	return []string{"backblaze"}
+}
+
+// Description returns a user-friendly description for this backend
+func (backend *StorageBackblaze) Description() string {
+	return "Backblaze Storage"
+}
+
+// AvailableSpace returns the free space on this backend
+func (backend *StorageBackblaze) AvailableSpace() (uint64, error) {
+	// Currently not supported
+	return 0, ErrAvailableSpaceUnknown
+}
+
+// LoadChunk loads a Chunk from backblaze
+func (backend *StorageBackblaze) LoadChunk(shasum string, part, totalParts uint) (*[]byte, error) {
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+	_, obj, err := backend.bucket.DownloadFileByName(fileName)
+	if err != nil {
+		return nil, err
+	}
+	data, err := ioutil.ReadAll(obj)
+	return &data, err
+}
+
+// StoreChunk stores a single Chunk from backblaze
+func (backend *StorageBackblaze) StoreChunk(shasum string, part, totalParts uint, data *[]byte) (size uint64, err error) {
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+
+	buf := bytes.NewBuffer(*data)
+	metadata := make(map[string]string)
+	i, err := backend.bucket.UploadFile(fileName, metadata, buf)
+	filesize := backblaze.File(*i)
+	return uint64(filesize.ContentLength), err
+}
+
+// LoadSnapshot loads a snapshot
+func (backend *StorageBackblaze) LoadSnapshot(id string) ([]byte, error) {
+	_, obj, err := backend.bucket.DownloadFileByName(id)
+	if err != nil {
+		return nil, ErrSnapshotNotFound
+	}
+	return ioutil.ReadAll(obj)
+}
+
+// SaveSnapshot stores a snapshot
+func (backend *StorageBackblaze) SaveSnapshot(id string, data []byte) error {
+	buf := bytes.NewBuffer(data)
+	metadata := make(map[string]string)
+	_, err := backend.bucket.UploadFile(id, metadata, buf)
+	return err
+}
+
+// InitRepository creates a new repository
+func (backend *StorageBackblaze) InitRepository() error {
+	var placeholder []byte
+	buf := bytes.NewBuffer(placeholder)
+
+	// Creating the files on backblaze
+	metadata := make(map[string]string)
+
+	if _, err := backend.bucket.UploadFile(backend.chunkFile, metadata, buf); err != nil {
```

Comment:
The chunkFile and snapshotFile seem to serve no purpose. They'll always be just empty files.